File: node/zkgroup/auth/AuthCredentialResponse.ts
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import ByteArray from '../internal/ByteArray';
import NativeImpl from '../../NativeImpl';
export default class AuthCredentialResponse extends ByteArray {
  static SIZE = 361;

  constructor(contents: Buffer) {
    super(contents, AuthCredentialResponse.SIZE, true);
    NativeImpl.AuthCredentialResponse_CheckValidContents(contents);
  }
}

File: java/shared/java/org/signal/libsignal/zkgroup/auth/AuthCredentialResponse.java
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.libsignal.zkgroup.auth;

import org.signal.libsignal.zkgroup.InvalidInputException;
import org.signal.libsignal.zkgroup.internal.ByteArray;
import org.signal.libsignal.internal.Native;

public final class AuthCredentialResponse extends ByteArray {
  public AuthCredentialResponse(byte[] contents) throws InvalidInputException {
    super(contents);
    Native.AuthCredentialResponse_CheckValidContents(contents);
  }
}
